### ISSUE TITLE
queryStringParams returns list of Strings

### ISSUE THREAD

**Issue by aldilaff** [awslabs/aws-sam-cli#404]
**Description:**
We just upgraded to SAM CLI 0.3.0 and noticed a change with queryStringParams where the query parameter is mapping to an array of a strings instead of a string. For example:

0.2.11 mapping: `{"name":"jason"}` 

0.3.0 mapping: `{'name': [u'jason']}`. 

We are using `aws-serverless-java-container-jersey` from https://github.com/awslabs/aws-serverless-java-container and it seems to break when trying to map the query parameter to a string when our endpoint is defined as follows:
`public Response getHelloWorld(@QueryParam("name") String name)`

We are not sure if SAM CLI is supposed to return a list of strings for query parameters or just a single string? Looking at the tests it seems that the expected behavior would be that each query parameter maps to a string and not a list of strings. test_api_event() in test_events.py has an expected_event where: `'queryStringParameters': {'foo': 'bar'}`

**Steps to reproduce the issue:**
1. Create a proxy integration lambda that prints out the inputStream
2. Call the new API endpoint with a query parameter and a value

**Observed result:**
The query parameter is mapped to a list containing the value given

**Output of `sam --version`**:
SAM CLI, version 0.3.0

**Comment by tlibey-panopto** [comment]
I am also having this issue. It is breaking existing unit tests. 

Details:
Our live lambda functions are still behaving as expected:
        "queryStringParameters": { "var": "val" },
but the same event in sam local is coming through as
        "queryStringParameters": {"var": ["val"]}

python 2.7. Just upgraded to aws-sam-cli-0.3.0 today. Is this a game breaking lambda change that is on its way? or just a aws-sam issue?

**Comment by nilobarp** [comment]
This update broke my entire app. All the `queryStringParameters` are getting mapped to list of strings instead of being a string.

**Comment by jfuss** [comment]
Working on a patch now. Will update when I have a PR out for this. Appologizes on the regression.

**Comment by jfuss** [comment]
Fix is out for review. Updating labels.

**Comment by mrchops1024** [comment]
I just want to add that this completely breaks dotnetcore. Passing any query string parameter at all results in an exception while invoking the handler:

`Newtonsoft.Json.JsonReaderException: Unexpected character encountered while parsing value: [. Path 'queryStringParameters`...

I just tested the fix from @jfuss and it appears to be working so far.

**Comment by mrchops1024** [comment]
@jfuss That's exactly what I did and so far there are no issues. Thanks.

**Comment by jonpierce** [comment]
I also encountered this issue after upgrading to 0.3.0. Glad to see there's an open PR that addresses it. Any ETA on when the PR will be merged and a new release cut with the fix? Thanks!

**Comment by jfuss** [comment]
@jonpierce I do not have an ETA yet. Going to poke some folks today in hopes to at least get it merged. I don't expect us to do a release right after that is merged though. There are a couple other regressions/bugs we need to fix and I think getting those into one release is best. 

We are still working on some of these process related things (cadence/criteria for a release is one of them). Once I have some clarity on that, I will update this issue so folks can at least understand when it will happen. Hopefully that helps.

**Comment by jonpierce** [comment]
@jfuss Cool, no worries. I've installed your branch locally and can just use that until then. Thanks.

**Comment by austinlparker** [comment]
As a heads-up, this bug also completely breaks golang. Trying to pass any query parameter results in the following error:
```{
  "errorMessage": "json: cannot unmarshal array into Go struct field APIGatewayProxyRequest.queryStringParameters of type string",
  "errorType": "UnmarshalTypeError"
}```

I can confirm that @jfuss patch fixes the problem. Would love to get an ETA on release...

**Comment by jfuss** [comment]
This was merged into develop. Putting into stage/accepted until a release is cut.

**Comment by jfuss** [comment]
This was released in [v0.4.0](https://github.com/awslabs/aws-sam-cli/releases/tag/v0.4.0)

Closing.
